### PR TITLE
Add autofill field comments

### PR DESCRIPTION
**Pull request by trmartin4**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

In anticipation of future changes, we want to develop or understanding of our Autofill logic.  I've added JSDoc comments on all functions in `autofill.js` as well as comments on each property of the `AutofillField` and `AutofillForm` classes, as well as the `AutofillPageDetails` class.

No code changed with this PR - it is just adding documentation so it is easier to understand what the functions are doing by hovering over them in VS Code and seeing the comments.

## Code changes

- **autofill.js:** Added comments on all function headers
- **autofillField.ts:** Added comments on all properties
- **autofillForm.ts:** Added comments on all properties
- **autofillPageDetails.ts:** Added comments on important properties

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/browser/src/content/autofill.js [hunk 1]
```diff
@@ -413,20 +464,35 @@
           return quebo;
       }
 
-      // clean up the text
+      /**
+       * Clean up the string `s` to non-printable characters and trim it.
```
Comment (eliykat) [inline apps/browser/src/content/autofill.js:468]:
```suggestion
       * Clean up the string `s` to remove non-printable characters and whitespace.
```

(I assume)

File: apps/browser/src/content/autofill.js [hunk 2]
```diff
@@ -493,7 +569,11 @@
           return theEl === document;
       }
 
-      // is a dom element "viewable" on screen?
+      /**
+       * Determine if the element is "viewable" on the screen
```
Comment (eliykat) [inline apps/browser/src/content/autofill.js:573]: I'd love it if you could clarify what "visible" vs. "viewable" means between these two functions. I've come across it before but never looked into it deeply.

File: apps/browser/src/content/autofill.js [hunk 3]
```diff
@@ -755,7 +849,12 @@
           return el ? (fillTheElement(el, op), [el]) : null;
       }
 
-      // do a fill by query operation
+      /**
+       * Do a fill by query operation
```
Comment (eliykat) [inline apps/browser/src/content/autofill.js:853]:
```suggestion
       * Find all elements matching `query` and fill them using the value `op` from the fill script
```

**Review by eliykat — CHANGES_REQUESTED** [review]
Amazing work @trmartin4! Thank you for sharing what you've figured out.

File: apps/browser/src/content/autofill.js (hunk 2 above)
Comment (trmartin4) [reply apps/browser/src/content/autofill.js]: I've added some more detailed documentation.  Let me know what you think.  It required quite a bit of testing to try to understand what it's doing, but I think I've got a pretty good idea and was able to convey that in the comments.

File: apps/browser/src/autofill/content/autofill.js [hunk 4]
```diff
@@ -470,37 +542,52 @@
...
 
-      // is a dom element visible on screen?
+      /**
+       * Determine if the element is visible.
+       * Visible is define as not having `display: none` or `visibility: hidden`.
+       * @param {HTMLElement} el 
+       * @returns {boolean} Returns `true` if the element is visible and `false` otherwise
+       */
       function isElementVisible(el) {
+          // Store the element we're checking
```
Comment (eliykat) [inline apps/browser/src/autofill/content/autofill.js:552]: This comment is unnecessary

File: apps/browser/src/autofill/content/autofill.js [hunk 5]
```diff
@@ -512,30 +599,56 @@
...
+         
+          // We then use elementFromPoint to find the element at that point.
+          for (var pointEl = el.ownerDocument.elementFromPoint(xCoordinate, yCoordinate); pointEl && pointEl !== el && pointEl !== document;) {
+                // If the element we found is a label, and the element we're checking has labels
+                if (pointEl.tagName && 'string' === typeof pointEl.tagName && 'label' === pointEl.tagName.toLowerCase()
                   && el.labels && 0 < el.labels.length) {
+                    // Return true if the element we found is one of the labels for the element we're checking.
+                    // This means that the element we're looking for is considered viewable
                   return 0 <= Array.prototype.slice.call(el.labels).indexOf(pointEl);
               }
```
Comment (eliykat) [inline apps/browser/src/autofill/content/autofill.js:636]: I appreciate that you've gone deeper on this, but the risk here is that you've started changing code which is very difficult to understand, difficult to test, and not covered by any unit or automated testing. While your changes are much more readable, I don't think this is worth it for the scope of this PR. I recommend restoring the previous code and just having a shorter/more general comment.

**Review by eliykat — CHANGES_REQUESTED** [review]

File: apps/browser/src/autofill/content/autofill.js (hunk 5 above)
Comment (trmartin4) [reply apps/browser/src/autofill/content/autofill.js:636]: I've removed any changes to the code and just combined the comments into one block at the top.

**Review by eliykat — APPROVED** [review]
Nice work!

**Pull request merged**